File: src/test/scala/com/botalista/basechange/BaseChangeTest.scala
package com.botalista.basechange

import BaseChange.alphaToDecimal
import BaseChange.decimalToAlpha

class BaseChangeTest extends munit.FunSuite:

  test("Alpha to decimal: zero")(assertEquals(alphaToDecimal("aaaa"), 0))
  test("Alpha to decimal: one")(assertEquals(alphaToDecimal("aaab"), 1))
  test("Alpha to decimal: end first digit")(
    assertEquals(alphaToDecimal("aaaz"), 24)
  )
  test("Alpha to decimal: start second digit")(
    assertEquals(alphaToDecimal("aaba"), 25)
  )
  test("Alpha to decimal: random")(assertEquals(alphaToDecimal("abcd"), 678))
  test("Alpha to decimal: laur")(assertEquals(alphaToDecimal("laur"), 172366))
  test("Alpha to decimal: last")(assertEquals(alphaToDecimal("zzzz"), 390624))

  test("Decimal to alpha: zero")(assertEquals(decimalToAlpha(0), "aaaa"))
  test("Decimal to alpha: one")(assertEquals(decimalToAlpha(1), "aaab"))
  test("Decimal to alpha: end first digit")(
    assertEquals(decimalToAlpha(24), "aaaz")
  )
  test("Decimal to alpha: start second digit")(
    assertEquals(decimalToAlpha(25), "aaba")
  )
  test("Decimal to alpha: random")(assertEquals(decimalToAlpha(678), "abcd"))
  test("Decimal to alpha: laur")(assertEquals(decimalToAlpha(172366), "laur"))
  test("Decimal to alpha: last")(assertEquals(decimalToAlpha(390624), "zzzz"))

File: src/test/java/com/botalista/basechange/BaseChangeTest.java
package com.botalista.basechange;

import static com.botalista.basechange.BaseChange.alphaToDecimal;
import static com.botalista.basechange.BaseChange.decimalToAlpha;

import junit.framework.TestCase;

/**
 * Unit test for simple App.
 */
public class BaseChangeTest extends TestCase {

  public void testZeroAtd() {
    assertEquals(0, alphaToDecimal("aaaa"));
  }

  public void testOneAtd() {
    assertEquals(1, alphaToDecimal("aaab"));
  }

  public void testEndFirstDigitAtd() {
    assertEquals(24, alphaToDecimal("aaaz"));
  }

  public void testStartSecondDigitAtd() {
    assertEquals(25, alphaToDecimal("aaba"));
  }

  public void testRandomAtd() {
    assertEquals(678, alphaToDecimal("abcd"));
  }

  public void testLaurAtd() {
    assertEquals(172366, alphaToDecimal("laur"));
  }

  public void testLastAtd() {
    assertEquals(390624, alphaToDecimal("zzzz"));
  }

  public void testZeroDta() {
    assertEquals("aaaa", decimalToAlpha(0));
  }

  public void testOneDta() {
    assertEquals("aaab", decimalToAlpha(1));
  }

  public void testEndFirstDigitDta() {
    assertEquals("aaaz", decimalToAlpha(24));
  }

  public void testStartSecondDigitDta() {
    assertEquals("aaba", decimalToAlpha(25));
  }

  public void testRandomDta() {
    assertEquals("abcd", decimalToAlpha(678));
  }

  public void testLaurDta() {
    assertEquals("laur", decimalToAlpha(172366));
  }

  public void testLastDta() {
    assertEquals("zzzz", decimalToAlpha(390624));
  }
}
